clientv3/watch.go

```diff
@@ -392,9 +395,11 @@ func (w *watcher) serveStream(ws *watcherStream) {
 				closing = true
 				break
 			}
-			newRev := wrs[0].Events[len(wrs[0].Events)-1].Kv.ModRevision
-			if newRev != ws.lastRev {
-				ws.lastRev = newRev
+			if len(wrs[0].Events) > 0 {
+				newRev := wrs[0].Events[len(wrs[0].Events)-1].Kv.ModRevision
+				if newRev != ws.lastRev {
+					ws.lastRev = newRev
```

Comment:
Yeah, `ws.Header.Revision` in the progress notification case would be the `newRev`